src_c/rect.c

```diff
@@ -695,6 +695,141 @@ pg_rect_collidelistall(pgRectObject *self, PyObject *args)
     return ret;
 }
 
+static SDL_Rect *
+pgRect_FromObjectAndKeyFunc(PyObject *obj, PyObject *keyfunc)
+{
+    SDL_Rect temp;
+    if (keyfunc == NULL) {
+        return pgRect_FromObject(obj, &temp);
```

Comment:
```suggestion
        SDL_Rect *ret = pgRect_FromObject(obj, &temp);
        if (!ret) {
            PyErr_SetString(
                PyExc_TypeError, "Sequence must contain rect or rect-like objects");
        }
        return ret;
```
The problem here is that `pgRect_FromObject` does not set a python error when it fails, and returns `NULL`. So we have to set python error here, so that the end result is that `pgRect_FromObjectAndKeyFunc` always sets python error on fail

Reply:
changed